File: src/main/java/com/itmo/microservices/shop/user/api/service/AuthService.java
package com.itmo.microservices.shop.user.api.service;

import com.itmo.microservices.shop.user.api.model.AuthenticationRequest;
import com.itmo.microservices.shop.user.api.model.AuthenticationResult;
import org.springframework.security.core.Authentication;

public interface AuthService {
    AuthenticationResult authenticate(AuthenticationRequest request);

    AuthenticationResult refresh(Authentication authentication);
}

File: src/main/kotlin/com/itmo/microservices/shop/demo/auth/api/service/AuthService.kt
package com.itmo.microservices.shop.demo.auth.api.service

import com.itmo.microservices.shop.demo.auth.api.model.AuthenticationRequest
import com.itmo.microservices.shop.demo.auth.api.model.AuthenticationResult
import org.springframework.security.core.Authentication

interface AuthService {
    fun authenticate(request: AuthenticationRequest): AuthenticationResult
    fun refresh(authentication: Authentication): AuthenticationResult
}